improvements-proposal.md

```diff
@@ -0,0 +1,7 @@
+1. Right now the data for the posts is coming from a json file. What changes would you have to make to the application if it came from an API? In what type of hook should you use to fetch the data and why? What other considerations would you have to make?
+
+  If the data for the posts came from an API, I would have to fetch the data in a useEffect() hook in the BlogList component and then set it to a state variable. Ideally this state would be handled globally with some sort of context provider like the useContext() hook or some state management library like Redux. I'd also consider using Axios for the API requests, or an ORM tool like Sequelize to avoid having to write my endpoints using PostgreSQL.
```

Comment:
Great answer!
- How could you improve your current solution if the number of blog posts was really large (for example 10,000)?
- What are additional features you would consider to add now that the data is being requested from an API?
- Could you explain how an ORM tool would be used when fetching the data from the API?